src/Microsoft.Net.Http.Headers/NameValueHeaderValue.cs

```diff
@@ -320,6 +320,25 @@ private static int GetNameValueLength(string input, int startIndex, out NameValu
             }
         }
 
+        internal static NameValueHeaderValue Find(ObjectCollection<NameValueHeaderValue> values, string name)
+        {
+            Contract.Requires((name != null) && (name.Length > 0));
+
+            if ((values == null) || (values.Count == 0))
+            {
+                return null;
+            }
+
+            foreach (var value in values)
+            {
+                if (string.Compare(value.Name, name, StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    return value;
+                }
+            }
+            return null;
+        }
```

Comment:
Non-ordered stuff though `IList` would be bad; as its harder to get an access by indexer out of an api that it would be to change the jit to understand `foreach` enumerators via interface without boxing; and since that hasn't happened yet its probably quite hard ;-)